src/Tasks/Microsoft.NET.Build.Tasks/RemoveDuplicatePackageReferences.cs

```diff
@@ -0,0 +1,53 @@
+﻿// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.Collections.Generic;
+using Microsoft.Build.Framework;
+using Microsoft.Build.Utilities;
+using System.Globalization;
+using NuGet.ProjectModel;
+using NuGet.Packaging.Core;
+using NuGet.Versioning;
+
+namespace Microsoft.NET.Build.Tasks
+{
+    /// <summary>
+    /// Resolves the assemblies to be published for a .NET app.
+    /// </summary>
+    public class RemoveDuplicatePackageReferences : TaskBase
+    {
+        private readonly List<ITaskItem> _packageList = new List<ITaskItem>();
+
+        [Required]
+        public ITaskItem[] InputPackageRefernces { get; set; }
+
+        /// <summary>
+        /// Unique package references
+        /// </summary>
+
+        [Output]
+        public ITaskItem[] UniquePackageRefernces
+        {
+            get { return _packageList.ToArray(); }
+        }
+
+
+        protected override void ExecuteCore()
+        {
+            var listofPackages = new HashSet<PackageIdentity>();
```

Comment:
`listofPackages` is misleading since it is a hash set.  Simply `packages` would work, I think.